### PR TITLE
add documentation about automating release using krew-release-bot

### PR DESCRIPTION
**Pull request by rajatjindal**

### CODE REVIEW

File: docs/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
 This helps users and maintainers to easily identify which version of the plugin
 they have installed.
 
+### Automating the releasing of new version
+
+You can use `github actions` to automate the release process. e.g. You can use
```
Comment (ahmetb) [inline docs/DEVELOPER_GUIDE.md]: Don’t put github actions in code font?

File: docs/DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
...
+`go-releaser` (or equivalent for other languages) for creation of new release and
```
Comment (ahmetb) [inline docs/DEVELOPER_GUIDE.md]: Don’t mention goreleaser it is not relevant here.

File: docs/DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
 This helps users and maintainers to easily identify which version of the plugin
 they have installed.
 
+### Automating the releasing of new version
```
Comment (ahmetb) [inline docs/DEVELOPER_GUIDE.md]: ## Automate releasing new versions on krew-index

File: docs/DEVELOPER_GUIDE.md (hunk 1 above)
Comment (ahmetb) [reply docs/DEVELOPER_GUIDE.md]: No need for e.g. here. You aren’t giving an example.

File: docs/DEVELOPER_GUIDE.md [hunk 4]
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
...
+
+You can use `github actions` to automate the release process. e.g. You can use
+`go-releaser` (or equivalent for other languages) for creation of new release and 
+[`krew-release-bot`](https://github.com/rajatjindal/krew-release-bot) to bump the 
+version in `krew-index` repo. Refer to the [documentation here](https://github.com/rajatjindal/krew-release-bot#basic-setup) for an example.
```
Comment (ahmetb) [inline docs/DEVELOPER_GUIDE.md]: No need for #target link. 
Just link to your repo home. It should be concise enough for someone to understand how it works in 60 seconds or so. :)

File: docs/DEVELOPER_GUIDE.md [hunk 5]
```diff
@@ -333,5 +333,10 @@ Ideally, the `version` specified should match the release tag of the plugin.
 This helps users and maintainers to easily identify which version of the plugin
 they have installed.
 
+### Automating the releasing of new version
+
+You can use github actions to publish new release of your krew plugin. `krew-release-bot`
+is a github action to automatically bump the version in `krew-index` repo. Refer to the [documentation here](https://github.com/rajatjindal/krew-release-bot) for details.
```
Comment (ahmetb) [inline docs/DEVELOPER_GUIDE.md]: 80 char line wraps pls :)

please capitalize as GitHub Actions.

File: docs/DEVELOPER_GUIDE.md (hunk 4 above)
Comment (rajatjindal) [reply docs/DEVELOPER_GUIDE.md]: done

File: docs/DEVELOPER_GUIDE.md (hunk 3 above)
Comment (rajatjindal) [reply docs/DEVELOPER_GUIDE.md]: done

File: docs/DEVELOPER_GUIDE.md (hunk 2 above)
Comment (rajatjindal) [reply docs/DEVELOPER_GUIDE.md]: done

File: docs/DEVELOPER_GUIDE.md (hunk 1 above)
Comment (rajatjindal) [reply docs/DEVELOPER_GUIDE.md]: done

File: docs/DEVELOPER_GUIDE.md (hunk 5 above)
Comment (rajatjindal) [reply docs/DEVELOPER_GUIDE.md]: sure, done

File: docs/DEVELOPER_GUIDE.md (hunk 3 above)
Comment (corneliusweig) [reply docs/DEVELOPER_GUIDE.md]: The latest version I see still says

> Automating the releasing of new version

@rajatjindal Can you check again?

File: docs/DEVELOPER_GUIDE.md [hunk 6]
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
 This helps users and maintainers to easily identify which version of the plugin
 they have installed.
 
+### Automating the releasing of new version
+
+You can use Github Actions to publish new release of your Krew plugin. 
+`krew-release-bot` is a github action to automatically bump the version in
```
Comment (corneliusweig) [inline docs/DEVELOPER_GUIDE.md]:
```suggestion
`krew-release-bot` is a GitHub Action to automatically bump the version in 
```
Better be consistent with regards to capitalization. IMO, `Action` is a name here and needs to have a capital letter too.

File: docs/DEVELOPER_GUIDE.md [hunk 7]
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
 This helps users and maintainers to easily identify which version of the plugin
 they have installed.
 
+### Automating the releasing of new version
+
+You can use Github Actions to publish new release of your Krew plugin. 
+`krew-release-bot` is a github action to automatically bump the version in 
+`krew-index` repo. 
+Refer to the [documentation here](https://github.com/rajatjindal/krew-release-bot) for details.
```
Comment (corneliusweig) [inline docs/DEVELOPER_GUIDE.md]:
```suggestion
Refer to the [krew-release-bot](https://github.com/rajatjindal/krew-release-bot) documentation for details.
```

File: docs/DEVELOPER_GUIDE.md (hunk 3 above)
Comment (rajatjindal) [reply docs/DEVELOPER_GUIDE.md]: yes, looks like I reverted that somehow, fixing again.

**Comment by rajatjindal** [comment]
> First of all, thanks for pushing this forward @rajatjindal !
> 
> Before we merge this, let's get a common understanding about whether `krew-release-bot` should be the recommended procedure for updates, as stated on your project landing page. I'd prefer to not have this statement there, because the _real_ goal is to have PRs with regular changesets, so that the approver-bot can handle them. Also, it's not clear who recommends this.



> First of all, thanks for pushing this forward @rajatjindal !
> 
> Before we merge this, let's get a common understanding about whether `krew-release-bot` should be the recommended procedure for updates, as stated on your project landing page. I'd prefer to not have this statement there, because the _real_ goal is to have PRs with regular changesets, so that the approver-bot can handle them. Also, it's not clear who recommends this.


sorry for the confusion it has caused. My intention was not to show recommendation from krew team, rather that statement was added when krew-release-bot had two ways of doing PR. (webhook & github-action) and out of those two github-action was the recommended way of doing PR.

I will remove that statement to avoid any further confusion. thanks for pointing out.

File: docs/DEVELOPER_GUIDE.md [hunk 8]
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
 This helps users and maintainers to easily identify which version of the plugin
 they have installed.
 
+### Automate releasing new versions on krew-index
+
+You can use Github Actions to publish new release of your Krew plugin. 
+`krew-release-bot` is a Github Action to automatically bump the version in 
+`krew-index` repo. 
+Refer to the [krew-release-bot](https://github.com/rajatjindal/krew-release-bot) for details.
```
Comment (corneliusweig) [inline docs/DEVELOPER_GUIDE.md]:
```suggestion
Refer to the [krew-release-bot](https://github.com/rajatjindal/krew-release-bot) documentation for details.
```

**Comment by corneliusweig** [comment]
:smiley: no worries. I'm open to also make this a krew-recommended upgrade method for version bumps, but it's a little early for that. Let's first to gather some more experience.

**Comment by ahmetb** [comment]
@corneliusweig tbh this is the "polished" experience compared to what it was before, and I suspect it's unlikely to change dramatically. I thought about image coming from a personal repo, but I think even that's OK for now. We just need to keep it static, as it works pretty decently at the moment.

**Comment by corneliusweig** [comment]
As long as no credentials or access tokens are passed to the action, the image can be hosted anywhere.

@rajatjindal when you made your final change, let's merge this!

**Comment by ahmetb** [comment]
Rather than the hosting of the image, the problem is to keep the API between container `<=>` service stable.

Since we don't use `:latest` in the image (and it would be dangerous to do so) any time we update the tag, we need to go update 20 other repos.

File: docs/DEVELOPER_GUIDE.md (hunk 8 above)
Comment (rajatjindal) [reply docs/DEVELOPER_GUIDE.md]: done

File: docs/DEVELOPER_GUIDE.md (hunk 7 above)
Comment (rajatjindal) [reply docs/DEVELOPER_GUIDE.md]: done

File: docs/DEVELOPER_GUIDE.md (hunk 6 above)
Comment (rajatjindal) [reply docs/DEVELOPER_GUIDE.md]: done

**Comment by rajatjindal** [comment]
Hi @corneliusweig/@ahmetb  thanks for all the feedback to get the wording right. I think I've addressed all the review comments now. If there is anything else that needs to be changed, I will be more than happy to do so.

thanks

**Comment by ahmetb** [comment]
Accidentally manually merged but
/lgtm
/approve

**Pull request merged**